demo/src/main/AndroidManifest.xml

```diff
@@ -43,7 +43,42 @@
     <activity android:name="com.google.android.exoplayer.demo.PlayerActivity"
         android:configChanges="keyboardHidden|orientation|screenSize"
         android:label="@string/application_name"
-        android:theme="@style/PlayerTheme"/>
+        android:theme="@style/PlayerTheme">
+
+        <intent-filter>
+            <action android:name="android.intent.action.VIEW"/>
+
+            <category android:name="android.intent.category.DEFAULT"/>
+            <category android:name="android.intent.category.BROWSABLE"/>
+
+            <data android:mimeType="video/*"/>
+        </intent-filter>
+
+        <intent-filter>
+            <action android:name="android.intent.action.VIEW"/>
+
+            <category android:name="android.intent.category.DEFAULT"/>
+            <category android:name="android.intent.category.BROWSABLE"/>
+
+            <data android:scheme="rtsp"/>
+        </intent-filter>
+
+        <intent-filter>
+            <!-- HTTP live support -->
```

Comment:
Given TYPE_OTHER is being used, would this actually work? You'd need the type to end up being TYPE_HLS. Also, why not support asset, file and content as well as http and https?